File: variables/perl/variables-Jeremy.pl
#!/usr/bin/env perl
#
# Introduction to variables in perl

$num  = 47;              # Scalar, holds one value at a time
@arr  = (1,2,3,4);       # Array, multiple value but must be the same type
%hash = ("one"=>1,       # Hash, holds a key and value
         "two"=>2); 

# Print of each variable (\n = newline)
print "num: $num\narr: @arr\nhash: $hash{one},$hash{two}\n";

push(@arr, 5);           # addes a variable to the end of an array
print "\n@arr\n";

pop(@arr);               # removes a variable from the end of an array
print "@arr\n";

# Makes an array that containes all the numbers inside the range
@count = (0 .. 50);
print "\nSize: ",scalar @count,".\n";  # Size of the array
print "Max Index: ",$#count,".\n";     # Prints the last index in the array

print "\n@count\n";
shift(@count);                         # Remove items from beginning of array
shift(@count);
print "\n@count\n";

unshift(@count,1);                     # Adds items to beginning of array
unshift(@count,0);
print "\n@count\n";

File: variables/ruby/variables-Jeremy.rb
#!/usr/bin/env ruby
#
# Introduction to variables in ruby

$var1 = 'Global'        # $ in the beginning makes a global variable
var2  = 'Local'         # Lowercase makes a local variable
VAR3  = 'Contant'       # All caps makes a constant variable

# Print out the type of variable it is
puts defined?($var1)
puts defined?(var2)
puts defined?(VAR3)
puts

puts "To access variables #{$var1}, #{var2}, and #{VAR3} you need \
to use the hash(#) symbol inside a double quoted string."

# Array with intergers
arr = [1,2,3,4,5]
puts arr[2] * arr[-1]   # Can do math operations in puts
                        # Arrays can be accessed forward and backwards
puts
puts arr                # Output the whole array without an index

print "This is the other output statement but it doesn't endline."
print " So it keeps on going"
puts

# When variable changes, it outputs a statement
trace_var:$var1, proc{puts "$var1 is now #{$var1}"}
$var1 = [5,4,3,2,1]